iot/contiki/README.md

```diff
@@ -0,0 +1,162 @@
+The Contiki Operating System for IoT
+====================================
+
+Contiki is an open source operating system that runs on tiny low-power
+microcontrollers and makes it possible to develop applications that
+make efficient use of the hardware while providing standardized
+low-power wireless communication for a range of hardware platforms.
+
+Contiki is used in numerous commercial and non-commercial systems,
+such as city sound monitoring, street lights, networked electrical
+power meters, industrial monitoring, radiation monitoring,
+construction site monitoring, alarm systems, remote house monitoring,
+and so on.
+
+For more information, see the Contiki website:
+
+[http://contiki-os.org](http://contiki-os.org)
+
+In this example, we will build a Singularity container with ContikiOS.
+We will base this example on the `contiki.def` (`def` for definition) file.
+First, we will go through an explanation of every section inside the `contiki.def` file:
+
+#### Environment section
+
+Here we define all the needed environment variables for ContikiOS to work fine.
+For this specific example the only needed variable is `CONTIKI` which is set to the
+path of the installation of contiki in the container.
+
+#### Help section
+
+Here we define the description of the container, in this case this container has
+ContikiOS as well as all the examples that come along including the Cooja simulator. We will test the functionality
+of this container by running a `Hello world` example below, by command line and by simulator.
+The help section also displays the correct way to run your container.
+
+#### Post section
+
+Here we define the needed dependencies to be installed in order to make ContikiOS run smoothly.
+One thing that is important is the set up of the dependencies for the specific architecture.
+The following example is based on a 64-bit architecture.
+
+Some other needed dependencies include the Java installation as well as the support for MSP430-based platforms and Java for Cooja.
+
+#### Runscript section
+
+Here all the needed commands to run the ContikiOS example are specified. More in detail, the Hello-world example provided from Contiki is run to know that the installation inside the container has been successful.
+
+
+How to build my Singularity ContikiOS container?
+================================================
+
+To build this container, you will need the following:
+
+ - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity).
+ - A text editor, like: `micro`, `vim` or `nano`.
+ - Root access.
+
+#### Build the container
+
+We will build and save the container into the `/contiki` folder. Make sure you have this folder before following the example. Go inside this folder in your host:
+
+```
+$ cd /contiki
+```
+
+And then you can build the container with a single command like this:
+
+```
+$ sudo singularity build contiki.sif contiki.def
+```
+
```

Comment:
Maybe add:

Or pull from the library:
```
$ singularity pull library://sylabs/examples/contiki:latest
```

Let me know when your ready and I'll sign, and push the container to the library.

Reply:
First the container needs to be tested by others, check if there's any troubleshooting problems with it then afterwards I think this could be added once there's the approval :+1: